localstack-core/localstack/services/apigateway/next_gen/execute_api/router.py

```diff
@@ -144,6 +145,13 @@ def create_not_found_response(api_id: str) -> Response:
         )
         return not_found
 
+    def vpc_endpoint_handler(
```

Comment:
I think this is a good way forward, as the shape of the URL is very different and doesn't have some of the parameters we're always expecting in a route, like `api_id`. Thanks for implementing it this way 🚀 and minimal changes with upstream dependencies 👍